test/rules/assertions/requireJsdoc.js

```diff
@@ -4,6 +4,124 @@
 
 export default {
   invalid: [
+    {
+      code: `
+          /** This is comment */
+          export interface Foo {
+            bar(): string;
+          }
+      `,
+      errors: [
+        {
+          message: 'Missing JSDoc comment.',
+          type: 'TSMethodSignature',
+        },
+      ],
+      options: [
+        {
+          contexts: [
+            'TSInterfaceDeclaration',
+            'TSMethodSignature',
+            'TSPropertySignature',
+          ],
+          publicOnly: {
+            ancestorsOnly: true,
+          },
+        },
+      ],
+      output: `
+          /** This is comment */
+          export interface Foo {
+            /**
+             *
+             */
+            bar(): string;
+          }
+      `,
+      parser: require.resolve('@typescript-eslint/parser'),
+    },
+    {
+      code: `
+          /** This is comment */
+          export interface Foo {
+            bar: string;
+          }
+      `,
+      errors: [
+        {
+          message: 'Missing JSDoc comment.',
+          type: 'TSPropertySignature',
+        },
+      ],
+      options: [
+        {
+          contexts: [
+            'TSInterfaceDeclaration',
+            'TSMethodSignature',
+            'TSPropertySignature',
+          ],
+          publicOnly: {
+            ancestorsOnly: true,
+            esm: true,
+          },
+        },
+      ],
+      output: `
+          /** This is comment */
+          export interface Foo {
+            /**
+             *
+             */
+            bar: string;
+          }
+      `,
+      parser: require.resolve('@typescript-eslint/parser'),
+    },
+    {
+      code: `
+      /**
+       * Foo interface documentation.
+       */
+      export interface Foo extends Bar {
+        /**
+         * baz method documentation.
+         */
+        baz(): void;
+
+        meow(): void;
+      }
+      `,
+      errors: [
+        {
+          message: 'Missing JSDoc comment.',
+        },
+      ],
+      options: [{
+        contexts: [
+          'TSMethodSignature',
+        ],
+        publicOnly: {
+          ancestorsOnly: true,
+        },
+      }],
+      output: `
+      /**
+       * Foo interface documentation.
+       */
+      export interface Foo extends Bar {
+        /**
+         * baz method documentation.
+         */
+        baz(): void;
+
+        /**
+         *
+         */
+        meow(): void;
+      }
+      `,
+      parser: require.resolve('@typescript-eslint/parser'),
+    },
     {
```

Comment:
How about a valid test too, e.g., to ensure a non-public one won't be flagged by `publicOnly`?

Reply:
There is no private in typescript `interface`.
But private method in exported class will always considered as exported, for even if the esm check consider the method is not exported, the variables map check can find the private method (cause the variable map construct dont consider any typescript feature)
In this PR I would prefer to focus on interface checking.

Reply:
Can make more PR about typescript classes, also #658 later